hardcoded.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+// Used to generate data for consts.php INSTALLER_TO_REPO_MINOR_VERSIONS
+// Copy .cow.pat.json from the current release into the directory this script is being run from (file is .gitignored)
+
+include 'consts.php';
+
+$filename = '.cow.pat.json';
+if (!file_exists($filename)) {
+    throw new \RuntimeException("Copy $filename into the directory this is being called from");
+}
+
+$versions = [];
+
+function repoName(string $name)
+{
+    $prefix = '';
+    if (!preg_match('#^(cwp/)#', $name) &&
+        !preg_match('#(/recipe-|/silverstripe-|/comment-notifications$)#', $name)
+    ) {
+        $prefix = 'silverstripe-';
+    }
+    if (preg_match('#(/agency-extensions|/starter-theme|/watea-theme)$#', $name)) {
+        $prefix = 'cwp-';
+    }
+    $name = str_replace('/agency-extensions', '/agencyextensions', $name);
```

Comment:
Still not protecting against `agency-extensions-something` if we have like a `silverstripe/agency-extensions-subsites` in the future.
It needs to either be `preg_replace` with a `$` at the end of the regex or just do it inside the above condition where that protection is already in place please.

Reply:
updated